domains/certificates/RegistryConnector/Worker/EventHandlers/MeasurementEventHandler.cs

```diff
@@ -0,0 +1,221 @@
+using DataContext;
+using DataContext.Models;
+using DataContext.ValueObjects;
+using Google.Protobuf;
+using MassTransit;
+using MassTransit.Courier.Contracts;
+using MeasurementEvents;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.PedersenCommitment;
+using ProjectOrigin.Registry.V1;
+using ProjectOrigin.WalletSystem.V1;
+using ProjectOriginClients;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using RegistryConnector.Worker.Exceptions;
+using RegistryConnector.Worker.RoutingSlips;
+
+namespace RegistryConnector.Worker.EventHandlers;
+
+public class MeasurementEventHandler : IConsumer<EnergyMeasuredIntegrationEvent>
+{
+    private readonly IEndpointNameFormatter endpointNameFormatter;
+    private readonly ProjectOriginOptions projectOriginOptions;
+    private readonly ApplicationDbContext dbContext;
+    private readonly ILogger<MeasurementEventHandler> logger;
+    private readonly IKeyGenerator keyGenerator;
+
+    public MeasurementEventHandler(IEndpointNameFormatter endpointNameFormatter, IOptions<ProjectOriginOptions> projectOriginOptions, ApplicationDbContext dbContext, ILogger<MeasurementEventHandler> logger, IKeyGenerator keyGenerator)
+    {
+        this.endpointNameFormatter = endpointNameFormatter;
+        this.projectOriginOptions = projectOriginOptions.Value;
+        this.dbContext = dbContext;
+        this.logger = logger;
+        this.keyGenerator = keyGenerator;
+    }
+
+    public async Task Consume(ConsumeContext<EnergyMeasuredIntegrationEvent> context)
```

Comment:
This method gets quite long. I think it makes sense to extract production and consumption into different private methods for readability

Reply:
We can do it but it results in a function with 6 parameters. I think that is many, so I actually like this better